guava/src/com/google/common/base/AbstractIterator.java

```diff
@@ -73,6 +73,13 @@ private boolean tryToComputeNext() {
   }
 
   @Override
+  @SuppressWarnings("return.type.incompatible") // warning is being issued for difference in super
+  // bound for next and the expected return type. If type is qualified with @Nullable annotation
```

Comment:
If hasNext() returns true, next will be non-null in case T is declared to be non-null type else it may or may not be null. And so will be the return type. Can this behaviour be expressed using annotations?

Reply:
I could only find the ways to confirm `@NonNull` type. I'm not sure how we can do this for ensuring type `T`. Can you please explain how it can be done?
Thanks!